core/src/server/rpc_module.rs

```diff
@@ -157,24 +173,23 @@ impl MethodCallback {
 	pub fn execute(
 		&self,
 		sink: &MethodSink,
+		conn_state: Option<ConnState>,
 		req: Request<'_>,
-		conn_id: ConnectionId,
 		claimed: Option<ResourceGuard>,
-		id_gen: &dyn IdProvider,
 	) -> MethodResult<bool> {
 		let id = req.id.clone();
 		let params = Params::new(req.params.map(|params| params.get()));
 
 		let result = match &self.callback {
 			MethodKind::Sync(callback) => {
 				tracing::trace!(
-					"[MethodCallback::execute] Executing sync callback, params={:?}, req.id={:?}, conn_id={:?}",
+					"[MethodCallback::execute] Executing sync callback, params={:?}, req.id={:?}, conn_state={:?}",
 					params,
 					id,
-					conn_id
+					conn_state
```

Comment:
I'd rather unwrap `conn_state` here than later (as per other comment), but it's still not clear to me why it's mandatory here but optional for async calls.

Reply:
I moved `conn_id, id_provider, close` to `ConnState` before we required to pass `conn_id` explictly here which is unused.

Thus, for HTTP (with connection state) you would just pass `execute(&sink, req, None, resources)` however I get your point that this is error prone and you can't really `unwrap` here but `subscriptions` must be initialized with `Some(conn_state)` if that makes sense.

What do say about `execute_with_state` and `execute`? However I guess we would need additional callback types then.... :grimacing: